staging/src/k8s.io/kube-aggregator/pkg/apiserver/openapi_aggregator.go

```diff
@@ -0,0 +1,253 @@
+	priority    int32
+}
+
+// byPriority can be used in sort.Sort to sort specs with their priorities.
+type byPriority []openAPISpecInfo
+
+func (a byPriority) Len() int      { return len(a) }
+func (a byPriority) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a byPriority) Less(i, j int) bool {
+	if a[i].priority != a[j].priority {
+		// Sort by priority, higher first
+		return a[i].priority > a[j].priority
+	} else {
+		// Sort by service name.
+		return a[i].serviceName < a[j].serviceName
+	}
+}
+
+// buildOpenAPISpec aggregates all OpenAPI specs.  It is not thread-safe.
+func (s *openAPIAggregator) buildOpenAPISpec() (specToReturn *spec.Swagger, err error) {
+	specToReturn, err = aggregator.CloneSpec(s.localDelegatesOpenAPISpec)
+	if err != nil {
+		return nil, err
+	}
+	if err := aggregator.MergeSpecs(specToReturn, s.aggregatorOpenAPISpec); err != nil {
+		return nil, fmt.Errorf("cannot merge local delegate spec with aggregator spec: %s", err.Error())
+	}
+	specs := []openAPISpecInfo{}
+	for name, specInfo := range s.openAPISpecs {
+		specs = append(specs, openAPISpecInfo{name, specInfo.spec, specInfo.priority})
+	}
+	sort.Sort(byPriority(specs))
+	for _, specInfo := range specs {
+		if err := aggregator.MergeSpecs(specToReturn, specInfo.spec); err != nil {
+			return nil, err
+		}
+	}
+	return specToReturn, nil
+}
+
+// updateOpenAPISpec aggregates all OpenAPI specs.  It is not thread-safe.
+func (s *openAPIAggregator) updateOpenAPISpec() error {
+	if s.openAPIService == nil {
+		return nil
+	}
+	specToServe, err := s.buildOpenAPISpec()
+	if err != nil {
+		return err
+	}
+	return s.openAPIService.UpdateSpec(specToServe)
+}
+
+// inMemoryResponseWriter is a http.Writer that keep the response in memory.
+type inMemoryResponseWriter struct {
+	header   http.Header
+	respCode int
+	data     []byte
+}
+
+func newInMemoryResponseWriter() *inMemoryResponseWriter {
+	return &inMemoryResponseWriter{header: http.Header{}}
+}
+
+func (r *inMemoryResponseWriter) Header() http.Header {
+	return r.header
+}
+
+func (r *inMemoryResponseWriter) WriteHeader(code int) {
+	r.respCode = code
+}
+
+func (r *inMemoryResponseWriter) Write(in []byte) (int, error) {
+	r.data = append(r.data, in...)
+	return len(in), nil
+}
+
+func (r *inMemoryResponseWriter) String() string {
+	s := fmt.Sprintf("ResponseCode: %d", r.respCode)
+	if r.data != nil {
+		s += fmt.Sprintf(", Body: %s", string(r.data))
+	}
+	if r.header != nil {
+		s += fmt.Sprintf(", Header: %s", r.header)
+	}
+	return s
+}
+
+func (s *openAPIAggregator) handlerWithUser(handler http.Handler, info user.Info) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		if ctx, ok := s.contextMapper.Get(req); ok {
+			s.contextMapper.Update(req, request.WithUser(ctx, info))
+		}
+		handler.ServeHTTP(w, req)
+	})
+}
+
+// Load openAPI spec from /swagger.json of given handler.
+func (s *openAPIAggregator) loadOpenAPISpec(handler http.Handler) (*spec.Swagger, error) {
```

Comment:
Handler means it gets an input and when it got it will return a result. Method loads spec and return it. I am not sure if load means side effect, but I can rename it to "get" or "retrieve" or "download"?